src/tests/ftest/get_remote_files.sh

```diff
@@ -0,0 +1,329 @@
+        if scp -r "${file}" "${2}"/"${archive_name}"; then
+            copied+=("${file}")
+            if ! rm -fr "${file}"; then
+                echo "  Error removing ${file}"
+                echo "    $(ls -al ${file})"
+                rc=1
+            fi
+        else
+            echo "  Failed to archive ${file} to ${2}"
+            rc=1
+        fi
+    done
+    echo "  The following files were archived:"
+    for file in "${copied[@]:-no files}"
+    do
+        echo "    ${file}"
+    done
+    return ${rc}
+}
+
+#######################################
+# Run cart_logtest.py against the logs.
+# Arguments:
+#   $1: local logs
+# Returns:
+#   0: All cart_logtest were successful
+#   1: Failure detected with cart_logtest
+#######################################
+run_cartlogtest() {
+    local rc=0
+    if [ -f "${CART_LOGTEST_PATH}" ]; then
+        # shellcheck disable=SC2045,SC2086
+        for file in $(ls -d ${1})
+        do
+            if [ -f ${file} ] && [[ ! ${file} == *".cart_logtest."* ]]; then
+                logtest_log="${file%.*}.cart_logtest.${file##*.}"
+                if ! ${CART_LOGTEST_PATH} ${file} > ${logtest_log} 2>&1; then
+                    echo "  Error: details in ${file}_cart_testlog"
+                    rc=1
+                fi
+            fi
+        done
+    else
+        echo "  Error: ${CART_LOGTEST_PATH} does not exist!"
+        rc=1
+    fi
+    return ${rc}
+}
+
+####### MAIN #######
+if [[ $# -eq 0 ]] ; then
+    usage
+    exit 1
+fi
+
+# Setup defaults
+ARCHIVE_DEST=""
+FILES_TO_PROCESS=""
+COMPRESS="false"
+VERBOSE="false"
+CART_LOGTEST="false"
+EXCLUDE_ZIP="false"
+THRESHOLD=""
+rc=0
+
+# Step through arguments
+while getopts "vhxzca:f:t:" opt; do
+    case ${opt} in
+        a )
+            ARCHIVE_DEST=$OPTARG
+            ;;
+        f )
+            FILES_TO_PROCESS=$OPTARG
+            ;;
+        t )
+            THRESHOLD=$OPTARG
+            ;;
+        c )
+            CART_LOGTEST="true"
+            ;;
+        z )
+            COMPRESS="true"
+            ;;
+        x )
+            EXCLUDE_ZIP="true"
+            ;;
+        v )
+            VERBOSE="true"
+            ;;
+        h )
+            usage
+            exit
+            ;;
+        ? )
+            echo "Unexpected argument: ${opt}!"
+            usage
+            exit
+            ;;
+    esac
+done
+shift "$((OPTIND -1))"
+
+if ! "${VERBOSE}"; then
+    # Redirect 'set -e' output to a log file if not verbose
+    script_name="${0##*/}"
+    log_file="/tmp/${script_name%%.*}_$(/bin/date +%Y%m%d_%H%M%S).log"
+    exec 2> "${log_file}"
+    echo "Logging ${script_name} output to ${log_file}"
+fi
+set -ex
+
+# Verify files have been specified and they exist on this host
+if ! check_files_input "${FILES_TO_PROCESS}"; then
+    if [[ $? -eq 1 ]]; then
+        echo "Please specify -f option."
+        exit 1
+    elif [[ $? -eq 2 ]]; then
```

Comment:
This should be checking the return status of `check_files_input `, not the last command.

Reply:
Fixed

Reply:
This is still not working correctly:
```
[2020-10-16T17:29:36.966Z] Running on wolf-101vm[1-6]: /usr/lib/daos/TESTING/ftest/get_remote_files.sh -z -a "wolf-101vm1:/var/tmp/ftest/avocado/job-results/latest/cart_logs" -f "/var/tmp/daos_testing/*/*log*" -c -t "1G"
[2020-10-16T17:29:36.966Z]     wolf-101vm[1-6]: rc=0, output: 
[2020-10-16T17:29:36.966Z]       Logging get_remote_files.sh output to /tmp/get_remote_files_20201016_172925.log
[2020-10-16T17:29:36.966Z]       Checking if files exceed 1G threshold ...
[2020-10-16T17:29:36.966Z]       Running /usr/lib/daos/TESTING/ftest/cart/cart_logtest.py ...
[2020-10-16T17:29:36.966Z]       Compressing files larger than 1M ...
[2020-10-16T17:29:36.966Z]       VM system detected, skipping compression ...
[2020-10-16T17:29:36.966Z]       Archiving logs to wolf-101vm1:/var/tmp/ftest/avocado/job-results/latest/cart_logs ...
[2020-10-16T17:29:36.966Z]         The following files were archived:
[2020-10-16T17:29:36.966Z]           no files
[2020-10-16T17:29:36.966Z]         The following files were archived:
[2020-10-16T17:29:36.966Z]           /tmp/get_remote_files_20201016_172925.log
```
It should look like this:
```
[2020-10-16T17:29:36.966Z] Running on wolf-101vm[1-6]: /usr/lib/daos/TESTING/ftest/get_remote_files.sh -z -a "wolf-101vm1:/var/tmp/ftest/avocado/job-results/latest/cart_logs" -f "/var/tmp/daos_testing/*/*log*" -c -t "1G"
[2020-10-16T17:29:36.966Z]     wolf-101vm[1-6]: rc=0, output: 
[2020-10-16T17:29:36.966Z]       Logging get_remote_files.sh output to /tmp/get_remote_files_20201016_172925.log
[2020-10-16T17:29:36.966Z]       No files matched /var/tmp/daos_testing/*/*log*. Nothing to do.
```
The https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-2029/59/artifact/Functional/job-2020-10-16T17.26-228b64e-io-llnl_mpi4py/cart_logs/get_remote_files_20201016_172925.wolf-101vm1.log/*view*/ indicates:
```
+ check_files_input '/var/tmp/daos_testing/*/*log*'
+ local rc=0
+ '[' -n '/var/tmp/daos_testing/*/*log*' ']'
+ ls -d '/var/tmp/daos_testing/*/*log*'
ls: cannot access /var/tmp/daos_testing/*/*log*: No such file or directory
+ rc=2
+ return 2
+ ret=0
+ [[ 0 -eq 1 ]]
+ [[ 0 -eq 2 ]]
```